src/Compiler/pars.fsy

```diff
@@ -3553,14 +3545,11 @@ constrPattern:
 atomicPatsOrNamePatPairs:
   | LPAREN namePatPairs rparen
       { let mParen = rhs2 parseState 1 3
-        let pats, m = $2
         let trivia = { ParenRange = mParen }
-        SynArgPats.NamePatPairs(pats, m, trivia), snd $2 }
+        SynArgPats.NamePatPairs($2, rhs parseState 2, trivia) }
 
   | atomicPatterns
-      { let mParsed = rhs parseState 1
-        let mAll = (mParsed.StartRange, $1) ||> unionRangeWithListBy (fun p -> p.Range)
```

Comment:
I don't know what the point of this was to be honest. It looks like it's equivalent to `lhs parseState`?

Reply:
It's similar but has important difference: if code is unfinished, then `lhs` may often include subsequent nodes range, which is wrong and is a big issue for us. The code above unions ranges of actually parsed things.

Reply:
Can you think of an example? I'll check whether there's been a change.

Reply:
Here's an example of this kind of issues: https://github.com/dotnet/fsharp/pull/12224.

I've been replacing using `lhs`/`rhs` with calculating the ranges like in the removed code for nodes where error recovery is expected:

For example, here the pattern range must end after the comma, but if using these functions, it may also include `=`:
```fsharp
let a, b, = ()
```

Reply:
The `SynPat.Tuple` spans all the way to `=` with and [without my changes](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYVuMACNBEMeIlgAvGAAUASjwFIsAB4AHKsmgUeAJwCusEAF8gA).

Reply:
Well, then it means that I, or somebody else, is going to fix it eventually. 🙂 But it'll require bringing calculating the range from the successfully parsed inner patterns and commas.

Reply:
Do you suggest I revert the parser changes? We'll lose the ability to provide accurate completions (a list containing only union case field names) for `| C  (first = f; | )`, which is not a huge deal, because merely have to type an extra letter.

Reply:
No, I'm just trying to answer the initial question. 🙂 And also saying that using `rhs`/`lhs` may create new issues when there's recovery in the inner rules, but we can fix them later if/when find them (e.g. when improving recovery further). The syntax test change (adding `)` to the range) looks good to me, thank you for this!

Reply:
Alright, thanks for reviewing and bouncing off ideas.

Reply:
@kerams Would it be possible to pass the `mParen` range from `atomicPatsOrNamePatPairs` and union the `constrPattern` ranges with it? That would reduce the changes and also make it easier to try to make recovery for missing `)` in future.